source/_drafts/Bugpost-Disconnects-and-Reconnects.md

```diff
@@ -0,0 +1,98 @@
+
+### Console Crash
+
+Imagine your console filled with the following two messages repeated thousands of times:
+```
+GET https://dash.akamaized.net/akamai/bbb_30fps/bbb_a64k/bbb_a64k_8.m4a net::ERR_INTERNET_DISCONNECTED
+VIDEOJS: WARN: Problem encountered with the current HLS playlist. Trying again since it is the final playlist.
+```
+
+When your console freezes, debugging can be tough. It's like trying to watch a video when the microwave is on. So we tackled that issue first.
+
+Looking at the error message, we were retrying the last rendition over and over. That is expected behavior, as we added that in https://github.com/videojs/videojs-contrib-hls/pull/1039 (a long time ago). However, DASH uses a different playlist loader than HLS, and we never added the same logic for reloading the final DASH playlists.
+
+After [adding the throttle](https://github.com/videojs/http-streaming/pull/277), debugging became much easier, and we were able to use the console once again and dig into why DASH didn't reconnect while HLS did.
+
+### Multiple Requests Multiple Problems
+
+The last problem proved to be the most difficult to debug. Initially, the thought was that demuxed content was the problem, as having a separate audio segment loader to load from the audio playlist could cause trouble when experiencing errors at the same time as the main segment loader tried to load from the video playlist. However, HLS can also have demuxed content, and after an example source proved to resume just fine, that approach was ruled out.
+
+But there were more than just two requests for DASH. By examining Chrome's network log, the DASH content, after a certain point, would stop requesting video segments, and only request audio segments. Further, the video segments included two requests: one for the init segment, and the other for the segment itself.
+
+Requests in VHS are managed by a module called `media-segment-request`, which is responsible for requesting everything required for a segment, whether it be the key, the init segment, the segment itself, or all of them, before letting the segment loader know that its requests are done. And for the video requests, the `media-segment-request` was never calling the done callback after the disconnect.
+
+It turned out that if we received one error, we would abort the other requests in that group, and we waited to call the `done` callback until those requests reported that they had been aborted. By the [standards](https://xhr.spec.whatwg.org/#the-abort()-method) though, the abort algorithm may not be run if the request was unsent. In this case, when the network was disconnected, the first request would report an error before the second had a chance to be sent, and so after we aborted the second request, we'd be stuck waiting for an error that would never come.
```

Comment:
Looks like `[standards](https://xhr.spec.whatwg.org/#the-abort()-method)` isn't displaying properly

Reply:
you can probably use a reference:

``` [standards][abort-method]```

and this at the bottom:
```[abort-method]: https://xhr.spec.whatwg.org/#the-abort()-method```

Reply:
Nice catch. Tried adding it at the bottom and it still had trouble. Encoded the parenthesis in the URL and it works.